File: src/main/java/org/mitallast/queue/common/events/EventBusModule.java
package org.mitallast.queue.common.events;

import com.google.inject.AbstractModule;

public class EventBusModule extends AbstractModule {
    @Override
    protected void configure() {
        bind(DefaultEventBus.class).asEagerSingleton();
        bind(EventBus.class).to(DefaultEventBus.class);
    }
}

File: src/main/java/org/mitallast/queue/common/events/EventBusModule.kt
package org.mitallast.queue.common.events

import com.google.inject.AbstractModule

class EventBusModule : AbstractModule() {
    override fun configure() {
        bind(DefaultEventBus::class.java).asEagerSingleton()
        bind(EventBus::class.java).to(DefaultEventBus::class.java)
    }
}
